pkg/services/provisioning/dashboards/vcs/provisioner.go

```diff
@@ -0,0 +1,214 @@
+			return err
+		}
+	}
+
+	return nil
+}
+
+// saveProvisionedDashboard creates the structures to save a dashboard and its provisioning info in store
+func (p *Provisioner) saveProvisionedDashboard(dashSvc dashboards.DashboardProvisioningService, obj vcs.VersionedObject, pastProvisioningInfo *models.DashboardProvisioning) error {
+	// Unmarshal the dashboard json
+	dash := &models.Dashboard{}
+	err := json.Unmarshal(obj.Data, dash)
+	if err != nil {
+		return err
+	}
+	path := getDashboardPath(obj.ID)
+
+	// Prepare info to save dashboard in store
+	dto := dashboards.SaveDashboardDTO{
+		OrgId:     dash.OrgId,
+		UpdatedAt: dash.Updated,
+		Message:   "Provisioned by " + ProvisionerName,
+		Overwrite: false,
+		Dashboard: dash,
+	}
+
+	if dto.Dashboard.Id != 0 {
+		dto.Dashboard.Data.Set("id", nil)
+		dto.Dashboard.Id = 0
+	}
+
+	if pastProvisioningInfo != nil {
+		dto.Dashboard.SetId(pastProvisioningInfo.DashboardId)
+	}
+
+	// Prepare info to save provisioning information in store
+	dp := &models.DashboardProvisioning{
+		ExternalId: path,
+		Name:       ProvisionerName,
+		Updated:    obj.Timestamp,
+		CheckSum:   obj.Version,
+	}
+
+	// Use the dashboard provisioning service to save the dashboard
+	p.log.Debug("saving new dashboard", "provisioner", ProvisionerName,
+		"file", path,
+		"folderId", dash.FolderId)
+
+	_, err = dashSvc.SaveProvisionedDashboard(&dto, dp)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// PollChanges calls the Provision method on a ticker
+func (p *Provisioner) PollChanges(ctx context.Context) {
+	ticker := time.NewTicker(time.Duration(int64(time.Second) * PollingPeriod))
+	for {
+		select {
+		case <-ticker.C:
+			p.Provision(ctx)
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// GetProvisionerResolvedPath returns the path of the config file for the provisioner
+// The VCS provisioner only has one path: vcs.Dashboard
+func (p *Provisioner) GetProvisionerResolvedPath(name string) string {
+	return fmt.Sprintf("%s/%s/", ProvisionerName, vcs.Dashboard)
+}
+
+// GetAllowUIUpdatesFromConfig returns whether provisioned dashboards for a given provisioner
+// can be updated from frontend or not. Since we handle saving to the VCS storage, it makes
+// sense to allow updates.
+func (p *Provisioner) GetAllowUIUpdatesFromConfig(name string) bool {
+	return true
+}
+
+// CleanUpOrphanedDashboards removes dashboards that are not in the VCS storage anymore
+func (p *Provisioner) CleanUpOrphanedDashboards() {
+	if p.VCS.IsDisabled() {
+		p.log.Warn("cannot clean up orphaned dashboards, VCS service is disabled")
+		return
+	}
+
+	vobjs, err := p.VCS.Latest(context.TODO(), vcs.Dashboard)
+	if err != nil {
+		p.log.Warn("cannot clean up orphaned dashboards", err)
+		return
+	}
+
+	dashSvc := dashboards.NewProvisioningService(p.Store)
+	provisionedDashboardRefs, err := getProvisionedDashboardsByPath(dashSvc, ProvisionerName)
+
+	for _, obj := range vobjs {
+		path := getDashboardPath(obj.ID)
```

Comment:
```suggestion
		// TODO: actually calls the Get function from the VCS service
		path := getDashboardPath(obj.ID)
```
Not sure about that. Anyway, it's too late to include it in the hackathon.